src/gatsby/src/components/park/parkDates.js

```diff
@@ -164,6 +171,37 @@ export default function ParkDates({ data, parkOperation }) {
       Object.values(openAccordions).every((isOpen) => isOpen)
   }, [openAccordions, subAreas.length])
 
+  const checkHash = useCallback(() => {
+    // Check hash in url
+    // if we find a matching parkSubArea, open that subArea accordion
+    if (typeof window !== "undefined") {
+      if (window.location.hash !== undefined && window.location.hash !== hash) {
+        const matchingSubArea = subAreas.find((subArea, idx) => {
+          if (window.location.hash === "#" + _.kebabCase(subArea.parkSubArea)) {
+            if (!openAccordions[idx]) {
+              setOpenAccordions((prev) => ({
+                ...prev,
+                [idx]: true,
+              }))
+            }
+            return true
+          }
+          return false
+        })
+        if (matchingSubArea) {
+          setHash(window.location.hash)
+        }
+      }
+    }
```

Comment:
I had some ideas for potentially simplifying this even more. Let me know what you think:

```suggestion
    const windowHash = window?.location?.hash
    if (!windowHash || windowHash === hash) return

    const matchingSubAreaIndex = subAreas.findIndex(
      (subArea) => windowHash === "#" + _.kebabCase(subArea.parkSubArea)
    )

    // If we don't find a matching parkSubArea, return
    if (matchingSubAreaIndex === -1) return

    if (!openAccordions[matchingSubAreaIndex]) {
      setOpenAccordions((prev) => ({
        ...prev,
        [matchingSubAreaIndex]: true,
      }))
    }

    setHash(windowHash)
```

I changed `find` to `findIndex`, since the index is what we want, and I moved the call to `setOpenAccordions` after the search, to avoid side effects in the `findIndex` loop.

The rest of it is just "early return" changes to reduce the number of indents. I know not everyone's a fan of those but I find it easier to keep track of what's going on with less nesting.

Hopefully the logic is still the same. Please take a look and let me know if any of these changes might improve readability!

Reply:
Yes, it works as expected! Thanks for your advice 🙏